app/components/Settings/Settings.react.js

```diff
@@ -334,7 +334,7 @@ function mapStateToProps(state) {
     } = state;
 
     const selectedConnectionId = tabMap[selectedTab];
-    const connectionsHaveBeenSaved = Boolean(selectedConnectionId);
+    const connectionsHaveBeenSaved = Boolean(selectedConnectionId) && connectRequests[selectedConnectionId];
```

Comment:
this logic might change depending on https://github.com/plotly/plotly-database-connector/pull/111/commits/4464119d468f27096f28608e100911401185b398#r91411335